### PR TITLE
[RFR] Bootstrap the project

### PR DESCRIPTION
**Pull request by Luwangel**
- [x] Add readme, gitignore, makefile
- [x] Setup next.js
- [x] Add a linter
- [x] Setup tests
- [x] Connect travis

### CODE REVIEW

**Review by djhi — CHANGES_REQUESTED** [review]
Enzyme is not configured so tests aren't bootstrapped yet. See https://github.com/airbnb/enzyme#installation

File: __tests__/pages/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import Index from '../../pages/Index';
+import { shallow } from 'enzyme';
+
+import renderer from 'react-test-renderer';
+
+describe('Index', () => {
+    it('should render Index correctly', () => {
```
Comment (Kmaschta) [inline __tests__/pages/index.spec.js]: Your test doesn't match your test phrase. `it should render without error` ?

File: __tests__/pages/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        renderer.create(<Index />);
+    });
+
+    it('should contain the main title', () => {
+        const app = shallow(<Index />);
+        expect(app.find('h1').text()).toEqual('Welcome to the 15 puzzle game!');
+    });
+
+    it('should match with snapshot', () => {
```
Comment (Kmaschta) [inline __tests__/pages/index.spec.js]: It should render correctly.. ?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.PHONY: help install run start test lint format
+
+help: ## Print all commands (default)
+	@grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+####### BUILD #######
+
+install: ## Install dependencies
```
Comment (floo51) [inline Makefile:10]: I cloned the repository, I made sure I had npm installed as in the README and launched `make install` (which isn't documented), it didn't work:

image

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	npm run build
+
+####### RUN #######
+
+run: ## Run the 15-puzzle isomorphic app
+	npm run start
```
Comment (floo51) [inline Makefile]: My other project app also runs on 3000, can you let me specify port through an environment variable ?
Something like `PORT=3001 make run` would be great !

**Review by floo51 — CHANGES_REQUESTED** [review]
Not sure why, but all I get is a 404 when launching (?)

File: Makefile (hunk 3 above)
Comment (Luwangel) [reply Makefile:10]: 404 explanation: index.js in pages should be in lowercase.

**Review by floo51 — APPROVED** [review]

**Review by djhi — APPROVED** [review]

**Pull request merged**